gist.el

```diff
@@ -240,25 +248,46 @@ Copies the URL into the kill ring."
     (mark-inactive (gist-buffer-private))))
 
 ;;;###autoload
-(defun gist-list (&optional force-reload background)
-  "Displays a list of all of the current user's gists in a new buffer."
-  (interactive "P")
+(defun gist-list-user (username &optional force-reload background)
+  "Displays a list of a user's gists in a new buffer.  When called from
+  a program, pass a blank string as the username to view the user's own
+  gists, or nil for the username and a non-nil value for force-reload to
+  reload the gists for the current buffer."
+  (interactive
+   (let ((username (read-from-minibuffer "GitHub user: " nil nil nil
+                                          'gist-user-history))
+         (force-reload (equal current-prefix-arg '(4))))
+     (list username force-reload)))
   ;; if buffer exists, it contains the current gh profile
   (let* ((gh-profile-current-profile (or gh-profile-current-profile
                                          (gh-profile-completing-read)))
-         (bufname (format "*%s:gists*" gh-profile-current-profile))
-         (api (gist-get-api nil)))
+         (bufname (if (null username)
+                      (if (not (equal major-mode 'gist-list-mode))
+                          (error "Current buffer isn't a gist-list-mode buffer")
+                        (buffer-name))
+                    (format "*%s:%sgists*"
+                            gh-profile-current-profile
+                            (if (string= "" username)
+                                ""
+                              (format "%s's-" username)))))
+         (api (gist-get-api nil))
+         (username (or (and (null username) gist-list-buffer-user)
+                       (and (not (or (null username)
+                                     (string= "" username)))
+                            username)
+                       (gh-api-get-username api))))
     (when force-reload
       (pcache-clear (oref api :cache))
-      (or background (message "Retrieving list of your gists...")))
+      (or background (message "Retrieving list of gists...")))
     (unless (and background (not (get-buffer bufname)))
-      (let ((resp (gh-gist-list api)))
+      (let ((resp (gh-gist-list api username)))
         (gh-url-add-response-callback
          resp
          (lexical-let ((buffer bufname))
            (lambda (gists)
              (with-current-buffer (get-buffer-create buffer)
-               (gist-lists-retrieved-callback gists background)))))
+               (setq gist-list-buffer-user username)
+               (gist-lists-retrieved-callback gists username background)))))
```

Comment:
gist-list-buffer-user is a buffer-local variable that stores the GitHub user for a gist list buffer (so that gist-list-user knows where to reload the list from).